drivers/overlay/overlay.go

```diff
@@ -882,6 +882,24 @@ func (d *Driver) UpdateLayerIDMap(id string, toContainer, toHost *idtools.IDMapp
 	if err := idtools.MkdirAs(diffDir, 0755, rootUID, rootGID); err != nil {
 		return err
 	}
+
+	if rootUID != 0 {
+		parent := filepath.Dir(diffDir)
+		for parent != "/" {
+			fi, err := os.Stat(parent)
+			if err != nil {
+				return err
+			}
+
+			const desiredMask = 0711
+			if fi.Mode()&desiredMask != desiredMask {
+				if err = os.Chmod(parent, fi.Mode()|desiredMask); err != nil {
+					return err
```

Comment:
Will this err give us enough information on which file/directory had the issue?  Could/should you wrap it with a little verbiage saying what parent was set to when the failure occurred.

Reply:
Yes that would make sense to wrap the err.

Reply:
I've pushed a new version with the Wrapf